### PR TITLE
Adapt the new version to UniProt

### PR DESCRIPTION
**Pull request by Swaathik**
- Fixed colouring for AlphaFold and added a new colour theme for Alphamissense
- Fixed the alignment of Nightingale-sequence-heatmap when used within the manager and adjusted the color-scale to match the rest.
- Added PTM residue in features part of Nightingale-track
- Migrated protvista-filter to Nightingale v5

### CODE REVIEW

File: packages/nightingale-structure/src/AlphaMissenseColorTheme.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        const cellContents = row.match(cellSplitter);
+        if (!cellContents)
+          throw new Error(
+            `FormatError: cannot parse "${row}" as a mutation (should look like Y123A)`
+          );
+        const seqId = +cellContents[2];
+        (scores[seqId - 1] ??= []).push(+cellContents[4]);
+      }
+
+      for (let i = 0, _i = residueRowCount; i < _i; i++) {
```
Comment (dlrice) [inline packages/nightingale-structure/src/AlphaMissenseColorTheme.ts:68]:
```suggestion
      for (let i = 0; i < residueRowCount; i++) {
```

File: packages/nightingale-structure/src/AlphaMissenseColorTheme.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        const averageScore = getAverage(scores[residueIndex[i]]);
+        map.set(i, averageScore);
+      }
+      return { value: map };
```
Comment (dlrice) [inline packages/nightingale-structure/src/AlphaMissenseColorTheme.ts:72]:
```suggestion
```

File: packages/nightingale-structure/src/af-confidence/prop.ts [hunk 3]
```diff
@@ -41,7 +41,10 @@ type AfConfidence = PropertyWrapper<
 export const DefaultServerUrl = "";
 
 export const isApplicable = (model?: Model): boolean => {
-  return !!model && Model.isFromPdbArchive(model);
+  if (model?.entryId.startsWith('AF')) {
```
Comment (dlrice) [inline packages/nightingale-structure/src/af-confidence/prop.ts:44]: Could so something like:

```
export const isApplicable = (model?: Model): boolean => !!model?.entryId.startsWith('AF')
```

File: packages/nightingale-track/src/nightingale-track.ts [hunk 4]
```diff
@@ -289,45 +316,103 @@ class NightingaleTrack extends withManager(
...
+          // It is placed in the middle of a single bandwidth
+          `translate(${
+            this.getXFromSeqPosition(
+              Number(f.feature.start) + Number(f.position) - 1
+            ) +
+            this.getSingleBaseWidth() / 4
+          },
+          ${this.layoutObj?.getFeatureYPos(f.feature)})`
+      )
+      .attr("fill", "#06038D")
```
Comment (dlrice) [inline packages/nightingale-track/src/nightingale-track.ts:408]: Add this to existing `config.ts` file

File: packages/nightingale-track/src/nightingale-track.ts [hunk 5]
```diff
@@ -360,14 +445,33 @@ class NightingaleTrack extends withManager(
...
             ),
-          [],
-        ),
+          []
+        )
+      );
+
+      const residueG = this.seqG.selectAll("g.residue-group").data(
+        this.#data.reduce(
+          (acc: unknown[], f) =>
```
Comment (dlrice) [inline packages/nightingale-track/src/nightingale-track.ts:460]:
```suggestion
          (feature_acc: unknown[], f) =>
```

File: packages/nightingale-track/src/nightingale-track.ts [hunk 6]
```diff
@@ -360,14 +445,33 @@ class NightingaleTrack extends withManager(
...
+            acc.concat([
+              f.ptms?.reduce(
+                (acc2: unknown[]) =>
```
Comment (dlrice) [inline packages/nightingale-track/src/nightingale-track.ts:463]:
```suggestion
                (ptm_acc: unknown[]) =>
```

File: packages/nightingale-track/src/nightingale-track.ts [hunk 7]
```diff
@@ -289,45 +316,103 @@ class NightingaleTrack extends withManager(
...
+        let ptmLength = 1;
+        // For longer proteins, the PTMs have to be shown prominent in the first look
+        if (this.length && this.length > 500) {
+          ptmLength = this.getSingleBaseWidth() < 4 ? 4 : 1;
+        }
+        return this.featureShape.getFeatureShape(
+          this.getSingleBaseWidth() / 2, // Halve the width of the ptm residue to distinguish between each other if one follows next closely
+          this.layoutObj?.getFeatureHeight() || 0,
+          ptmLength,
+          this.getShape(f)
```
Comment (dlrice) [inline packages/nightingale-track/src/nightingale-track.ts:393]:
```suggestion
        let ptmLength = 1;
        // For longer proteins, the PTMs have to be shown prominent in the first look
        if (this.length && this.length > 500) {
          ptmLength = this.getSingleBaseWidth() < 4 ? 4 : 1;
        }
        return this.featureShape.getFeatureShape(
          this.getSingleBaseWidth() / 2, // Halve the width of the ptm residue to distinguish between each other if one follows next closely
          this.layoutObj?.getFeatureHeight(f) || 0,
          ptmLength,
          this.getShape(f)
```
Can also extract into its own fn as it's used below in the refresh too.

File: packages/nightingale-filter/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+[
+  {
+    name: "filter_name",
+    type: {
+      name: "group_name",
+      text: "Group Label"
+    },
+    options: {
+      labels: ["Filter option label"],
+      colors: ["#333"] // Array so you can specify colour ranges
```
[the remark this answers is not in the document]
Comment (Swaathik) [reply packages/nightingale-filter/README.md]: I updated only the type change as the group name and label is clear in this rather than the TS type

**Review by dlrice — APPROVED** [review]

**Pull request merged**